api/src/Entity/NamedDeputy.php

```diff
@@ -511,4 +511,14 @@ public function addressHasChanged(OrgDeputyshipDto $dto)
              $this->getAddress5() !== $dto->getDeputyAddress5() ||
              $this->getAddressPostcode() !== $dto->getDeputyPostcode();
     }
+
+    public function nameHasChanged(OrgDeputyshipDto $dto): bool
+    {
+        if ($dto->deputyIsAnOrganisation()) {
```

Comment:
Although I know this is PHP I believe we can take some learnings from Go when writing `if/else` statements. In this scenario we're checking the happy path first and returning. If we do end up inside the `if` because we have an instant return we then know the code no longer executes any further in this function therefore making the `else` block redundant.

One of Go's other standards is you should always try and deal with the unhappy scenario first. In doing so I believe this function could be written as.

```
if (!$dto->deputyIsAnOrganisation()) {
    return $dto->getDeputyFirstname() !== $this->getFirstname() ||
            $dto->getDeputyLastname() !== $this->getLastname();
}

return return $dto->getOrganisationName() !== $this->getFirstname();
``` 

In my opinion I believe this just looks cleaner, and also we no longer have the extra block of code in the `else` statement as we deal with the unhappy path first and if we don't fall into it we just return the expected value.